### PR TITLE
[FSSDK-9155] fix(utils): logging correct error message in HTTP requests

### PR DESCRIPTION
**Pull request by pulak-opti**
### Summary
- Fix error log

## Issues
- [FSSDK-9155](https://jira.sso.episerver.net/browse/FSSDK-9155)

### CODE REVIEW

File: pkg/utils/requester_test.go [hunk 1]
```diff
@@ -190,12 +190,42 @@ func TestPostObj(t *testing.T) {
...
+
 func TestGetBad(t *testing.T) {
+	// Using a mockLogger to ensure we're logging the expected error message
+	mlog := &mockLogger{}
+	httpreq := NewHTTPRequester(mlog)
 
-	httpreq := NewHTTPRequester(logging.GetLogger("", ""))
-	_, _, _, err := httpreq.Get("blah12345/good")
-	_, ok := err.(*url.Error)
+	badURL := "blah12345/good"
```
Comment (mikechu-optimizely) [inline pkg/utils/requester_test.go:212]: NIT: "good" is a little misleading if this is a badURL 🤔

**Review by mikechu-optimizely — COMMENTED** [review]
LGTM. I added some NITs

File: pkg/utils/requester_test.go [hunk 2]
```diff
@@ -190,12 +190,42 @@ func TestPostObj(t *testing.T) {
 	assert.NotNil(t, err)
 }
 
+// mockLogger exists to check that the logged error is what we expected
```
[the remark this answers is not in the document]
Comment (yasirfolio3) [reply pkg/utils/requester_test.go:193]: I second this, the name is self explanatory so this isn't absolutely necessary.

File: pkg/utils/requester_test.go [hunk 3]
```diff
@@ -190,12 +190,42 @@ func TestPostObj(t *testing.T) {
...
+type mockLogger struct {
+	Errors []error
+}
+
+func (t *mockLogger) Debug(message string)   {}
+func (t *mockLogger) Info(message string)    {}
+func (t *mockLogger) Warning(message string) {}
+func (t *mockLogger) Error(message string, err interface{}) {
```
Comment (yasirfolio3) [inline pkg/utils/requester_test.go:201]:
```suggestion
func (m *mockLogger) Debug(message string)   {}
func (m *mockLogger) Info(message string)    {}
func (m *mockLogger) Warning(message string) {}
func (m *mockLogger) Error(message string, err interface{}) {
```

File: pkg/utils/requester_test.go [hunk 4]
```diff
@@ -190,12 +190,42 @@ func TestPostObj(t *testing.T) {
...
+	if err, ok := err.(error); ok {
+		t.Errors = append(t.Errors, err)
+	}
+}
+
 func TestGetBad(t *testing.T) {
+	// Using a mockLogger to ensure we're logging the expected error message
+	mlog := &mockLogger{}
```
Comment (yasirfolio3) [inline pkg/utils/requester_test.go:209]:
```suggestion
	mLogger := &mockLogger{}
```

**Review by yasirfolio3 — CHANGES_REQUESTED** [review]
Some minor changes suggested.

File: pkg/utils/requester_test.go (hunk 1 above)
Comment (pulak-opti) [reply pkg/utils/requester_test.go]: added this "htp:/ww.bad-url.fake/blah12345"

File: pkg/utils/requester_test.go [hunk 5]
```diff
@@ -190,12 +190,41 @@ func TestPostObj(t *testing.T) {
...
+
 func TestGetBad(t *testing.T) {
+	// Using a mockLogger to ensure we're logging the expected error message
+	mLogger := &mockLogger{}
+	httpreq := NewHTTPRequester(mLogger)
 
-	httpreq := NewHTTPRequester(logging.GetLogger("", ""))
-	_, _, _, err := httpreq.Get("blah12345/good")
-	_, ok := err.(*url.Error)
+	badURL := "htp:/ww.bad-url.fake/blah12345"
```
Comment (mikechu-optimizely) [inline pkg/utils/requester_test.go:211]: Is this better served as "http://ww.bad-url.fake/blah12345"

File: pkg/utils/requester_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pulak-opti) [reply pkg/utils/requester_test.go]: yeah, removed

File: pkg/utils/requester_test.go (hunk 5 above)
Comment (pulak-opti) [reply pkg/utils/requester_test.go]: agreed, updated

File: pkg/utils/requester_test.go (hunk 3 above)
Comment (pulak-opti) [reply pkg/utils/requester_test.go]: updated

File: pkg/utils/requester_test.go (hunk 4 above)
Comment (pulak-opti) [reply pkg/utils/requester_test.go]: updated

**Review by yasirfolio3 — APPROVED** [review]
LGTM

**Pull request merged**